### PR TITLE
wgsl: declare attribtues with @

### PR DESCRIPTION
**Pull request by kvark**
BREAKS EVERY PROGRAM
Matches https://github.com/gpuweb/gpuweb/pull/2517

### CODE REVIEW

File: tests/out/wgsl/800-out-of-bounds-panic-vert.wgsl [hunk 1]
```diff
@@ -32,8 +32,8 @@ fn main_1() {
     return;
 }
 
-[[stage(vertex)]]
-fn main([[location(0)]] position: vec2<f32>, [[location(1)]] color: vec4<f32>) -> VertexOutput {
+@stage(vertex) 
+fn main(@location(0) @ position: vec2<f32>, @location(1) @ color: vec4<f32>) -> VertexOutput {
```
Comment (mehmetoguzderin) [inline tests/out/wgsl/800-out-of-bounds-panic-vert.wgsl:36]: 35 seems to have extra space at end while 36 seems to contain wrong @ s

File: tests/out/wgsl/800-out-of-bounds-panic-vert.wgsl [hunk 2]
```diff
@@ -7,11 +7,11 @@ struct VertexPushConstants {
 };
 
 struct VertexOutput {
-    [[location(0)]] frag_color: vec4<f32>;
-    [[builtin(position)]] member: vec4<f32>;
+    @location(0) @ frag_color: vec4<f32>;
```
Comment (mehmetoguzderin) [inline tests/out/wgsl/800-out-of-bounds-panic-vert.wgsl:10]: Second @ seems wrong

File: tests/out/wgsl/800-out-of-bounds-panic-vert.wgsl [hunk 3]
```diff
@@ -7,11 +7,11 @@ struct VertexPushConstants {
...
+    @builtin(position) member: vec4<f32>;
 };
 
-[[group(0), binding(0)]]
+@group(0) @binding(0)
```
Comment (mehmetoguzderin) [inline tests/out/wgsl/800-out-of-bounds-panic-vert.wgsl:14]: Extra space at end (replace artifact?)

**Review by mehmetoguzderin — CHANGES_REQUESTED** [review]

File: tests/out/wgsl/800-out-of-bounds-panic-vert.wgsl (hunk 2 above)
Comment (mehmetoguzderin) [reply tests/out/wgsl/800-out-of-bounds-panic-vert.wgsl:10]: There are many other instances of these floating ` @ ` (space + at + space)

File: tests/out/wgsl/800-out-of-bounds-panic-vert.wgsl (hunk 3 above)
Comment (kvark) [reply tests/out/wgsl/800-out-of-bounds-panic-vert.wgsl:14]: I think it's fine to have that extra space

File: src/front/wgsl/mod.rs [hunk 4]
```diff
@@ -232,14 +229,12 @@ impl<'a> Error<'a> {
                         ExpectedToken::Integer => "unsigned/signed integer literal".to_string(),
                         ExpectedToken::Constant => "constant".to_string(),
                         ExpectedToken::PrimaryExpression => "expression".to_string(),
-                        ExpectedToken::AttributeSeparator => "attribute separator (',') or an end of the attribute list (']]')".to_string(),
                         ExpectedToken::FieldName => "field name or a closing curly bracket to signify the end of the struct".to_string(),
                         ExpectedToken::TypeAttribute => "type attribute ('stride') or an end of the attribute list (']]')".to_string(),
```
Comment (jimblandy) [inline src/front/wgsl/mod.rs:233]: There's a `]]` in this error message that probably doesn't belong there.

**Review by jimblandy — APPROVED** [review]

**Comment by jimblandy** [comment]
Approved, but do note the one error message that needs a tweak

File: src/front/wgsl/mod.rs (hunk 4 above)
Comment (kvark) [reply src/front/wgsl/mod.rs]: nicely spotted!

**Pull request merged**